### ISSUE TITLE
SyntaxError: Cannot use 'import.meta' outside a module

### ISSUE THREAD

**Issue by jimmed** [remix-run/remix#5644]
### What version of Remix are you using?

1.14.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Using a project based on the `cloudflare-workers` template:

1. Upgrade from Remix 1.13.0 to 1.14.0
2. Launch the app in local development mode (`npm run dev`)

### Expected Behavior

The app boots in Miniflare as normal

### Actual Behavior

On startup, Miniflare throws the following fatal errors:

```
[mf:err] /home/jim/src/ksoc/web-app/build/index.js:7882
    import.meta && import.meta.hot && import.meta.hot.accept("remix:manifest", async (newManifest) => {
           ^^^^

SyntaxError: Cannot use 'import.meta' outside a module
    at new Script (node:vm:100:7)
    at VMScriptRunner.runAsScript (/home/jim/src/ksoc/web-app/node_modules/@miniflare/runner-vm/src/index.ts:23:20)
    at VMScriptRunner.run (/home/jim/src/ksoc/web-app/node_modules/@miniflare/runner-vm/src/index.ts:88:12)
    at Miniflare.#reload (/home/jim/src/ksoc/web-app/node_modules/@miniflare/core/src/index.ts:775:42)
    at Miniflare.getPlugins (/home/jim/src/ksoc/web-app/node_modules/@miniflare/core/src/index.ts:1018:5)
    at createServer (/home/jim/src/ksoc/web-app/node_modules/@miniflare/http-server/src/index.ts:378:19)
    at startServer (/home/jim/src/ksoc/web-app/node_modules/@miniflare/http-server/src/index.ts:485:18)
    at main (/home/jim/src/ksoc/web-app/node_modules/miniflare/src/cli.ts:111:7)
[mf:err] Unhandled Promise Rejection: /home/jim/src/ksoc/web-app/build/index.js:7882
    import.meta && import.meta.hot && import.meta.hot.accept("remix:manifest", async (newManifest) => {
           ^^^^

SyntaxError: Cannot use 'import.meta' outside a module
    at new Script (node:vm:100:7)
    at VMScriptRunner.runAsScript (/home/jim/src/ksoc/web-app/node_modules/@miniflare/runner-vm/src/index.ts:23:20)
    at VMScriptRunner.run (/home/jim/src/ksoc/web-app/node_modules/@miniflare/runner-vm/src/index.ts:88:12)
    at Miniflare.#reload (/home/jim/src/ksoc/web-app/node_modules/@miniflare/core/src/index.ts:775:42)
    at Miniflare.getPlugins (/home/jim/src/ksoc/web-app/node_modules/@miniflare/core/src/index.ts:1018:5)
    at createServer (/home/jim/src/ksoc/web-app/node_modules/@miniflare/http-server/src/index.ts:378:19)
    at startServer (/home/jim/src/ksoc/web-app/node_modules/@miniflare/http-server/src/index.ts:485:18)
    at main (/home/jim/src/ksoc/web-app/node_modules/miniflare/src/cli.ts:111:7)
```

**Comment by jimmed** [comment]
This issue persists in 1.14.1

**Comment by m4rvr** [comment]
Happens for me to but only when using
`import { useDehydratedState } from 'use-dehydrated-state'`

When I use the same code from a local file e.g. `./use-dehydrated-state`, it magically works. 🤔

**Comment by aerojeyenth** [comment]
I encountered the same issue when trying to use `@clerk/remix` package in official cloudflare template from create-remix. 

This happened when I tried to follow the official documents at [https://clerk.com/docs/quickstarts/get-started-with-remix](https://clerk.com/docs/quickstarts/get-started-with-remix)

And the error occurs at 2nd step as follows:

![image](https://user-images.githubusercontent.com/6432625/226664685-fd81c567-2711-4967-8b12-3b7a48bb3a4a.png)

And also this looks like a known issue [https://github.com/remix-run/remix/pull/5259#discussion_r1131700486](https://github.com/remix-run/remix/pull/5259#discussion_r1131700486)

**Comment by jimmed** [comment]
This seems to be fixed as of @remix-run/* v1.15.0
